### PR TITLE
Alter JSON string parsing from shell

### LINKED ISSUE #2680: JSON parameters are not parsed correctly using CMD
It seems to be parsing this incorrectly and I'm not sure why?

```
az vmss extension set --publisher Microsoft.Azure.Extensions --version 2.0 --name CustomScript --resource-group vmss-test-1 --vmss-name myscaleset --settings '{"commandToExecute": "apt-get -y update && apt-get -y install nginx"}'

az: error: unrecognized arguments: apt-get -y update && apt-get -y install nginx}'
```

### PR DESCRIPTION
**Pull request by tjprescott**
Closes #2680.

Updates references where parameters required JSON strings to allow Python lists/dictionaries. Valid JSON is a valid Python list/dict, but the reverse is not true. On CMD, you would need to escape characters with something like the following:

`--parameters "{\"foo\": \"doo\"}"`

but certain escaped strings are simply not parsed correctly by the shell. For example, you cannot use:

`--parameters "{\"foo\": \"doo & boo\"}"`

This results in CMD terminating the command line at the &, resulting in errors. Previously, the only workaround would be to use the @file syntax, but there should be a way to specify purely on the command line.

This update will now allow the following:
`--parameters "{'foo': 'doo & boo'}"`

This simplifies the input on CMD greatly.

### CODE REVIEW

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 from azure.cli.command_modules.acs.mgmt_acs.lib \
     import AcsCreationClient as ACSClient
 # pylint: disable=too-few-public-methods,too-many-arguments,no-self-use,line-too-long
-from azure.cli.core._util import CLIError
+from azure.cli.core._util import CLIError, shell_safe_json_parse
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Just a thought. We should rename `_util` module to `util`. That fact that this module is imported from outside of the `azure.cli.core` module proves that it is not private module anymore. Leaving the name unchanged will cause ambiguity.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -140,6 +141,9 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
 register_cli_argument('vmss extension image', 'top', help='Return top number of records')
 register_cli_argument('vmss extension image', 'version', help='Extension version')
 
+register_cli_argument('vmss extension', 'settings', type=validate_file_or_dict)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: this will still persist the help text, right?

Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: The help text will still say JSON, and that is what is encouraged. However, for the CMD users (like moi) who need to work around this issue, they could also use the nearly identical dictionary syntax.

**Comment by tjprescott** [comment]
@troydai added unit test for shell_safe_json_parse

**Pull request merged**